projects/plugins/jetpack/modules/shortcodes/presentations.php

```diff
@@ -106,6 +106,9 @@ public function add_scripts() {
 			}
 
 			foreach ( $GLOBALS['posts'] as $p ) {
+				if ( ! $p ) {
```

Comment:
I'm afraid this might be too ambiguous. To avoid having PHP cast the variable to boolean in possibly unpredictable ways, how about we just check if this is not an object instead? We're expecting it to be an object anyway in the next line.

Reply:
Or just make line 112 be like `isset( $p->post_content ) && has_shortcode( $p->post_content, 'presentation' )`, since that's the only place it gets used in here.